controller/src/beerocks/master/son_master_thread.cpp

```diff
@@ -1527,7 +1527,46 @@ bool master_thread::handle_non_intel_slave_join(Socket *sd,
 {
     // TODO backhaul handling.
     // Multi-AP Agent doesn't say anything about the backhaul, so we have to rely on Intel Slave Join for that.
-    std::string backhaul_mac = "";
+    std::string backhaul_mac     = network_utils::get_random_mac();
+    std::string parent_bssid_mac = network_utils::ZERO_MAC_STRING;
+    std::string manufacturer(tlvwscM1->manufacturer(), tlvwscM1->manufacturer_length());
+
+    LOG(INFO) << "IRE generic Slave joined, sd=" << intptr_t(sd) << std::endl
+              << "    manufacturer=" << manufacturer << std::endl
+              << "    parent_bssid_mac=" << parent_bssid_mac << std::endl
+              << "    backhaul_mac=" << backhaul_mac << std::endl
+              << "    al_mac=" << bridge_mac << std::endl
+              << "    radio_identifier = " << radio_mac << std::endl;
+
+    //TODO might need to handle bssids of VAP nodes as well in this case
+    if (database.get_node_state(backhaul_mac) != beerocks::STATE_CONNECTED &&
+        database.get_node_state(backhaul_mac) != beerocks::STATE_CONNECTED_IP_UNKNOWN) {
```

Comment:
If you just generated the backhaul mac, it will not be in the DB, so why do you write this condition?
`database.get_node_state(backhaul_mac)` will return `"beerocks::STATE_MAX"` so the condition will always be true.